### PR TITLE
Feature/dev 707 kyc link

### PR DESCRIPTION
**Pull request by ThaBrad**
#### What kind of PR is this?:
/kind feature

#### What this PR does / why we need it:
Add correct KYC link and jumps to KYC Chatbot

#### Developer Checklist:
- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] Your UI implementation visually matched the rendered design*

### CODE REVIEW

File: shared/api/dfx/ApiService.ts [hunk 1]
```diff
@@ -76,32 +78,84 @@ export const getUserDetail = async (): Promise<UserDetail> => {
...
+  return await fetchFrom<UserDetailDto>(UserUrl, 'PUT', toUserDto(user as User)).then(fromUserDetailDto)
 }
 
 export const updateRefFee = async (fee: number): Promise<void> => {
   return await fetchFrom(UserUrl, 'PUT', { refFeePercent: fee })
 }
 
 // --- KYC --- //
-export const putKycData = async (data: KycData): Promise<void> => {
+export const putKycData = async (data: KycData, code?: string): Promise<KycInfo> => {
```
Comment (davidleomay) [inline shared/api/dfx/ApiService.ts:90]: As discussed, I would remove the legacy calls and switch to required code parameter.

File: shared/contexts/DFXAPIContextProvider.tsx [hunk 2]
```diff
@@ -80,6 +81,32 @@ export function DFXAPIContextProvider (props: PropsWithChildren<{}>): JSX.Elemen
...
+    await getActiveWebToken()
+      .catch(async () => {
+        // try login first
+        await activePairHandler({ network: networkName, addr: address })
+        return await getActiveWebToken()
+      })
+      .then(async (token) => {
+        if (token === undefined || token.length === 0) {
+          throw new Error('webToken is undefined')
+        }
```
Comment (davidleomay) [inline shared/contexts/DFXAPIContextProvider.tsx:94]: Should we extract this to a separate method as it is used multiple times?

File: shared/contexts/DFXAPIContextProvider.tsx [hunk 3]
```diff
@@ -80,6 +81,32 @@ export function DFXAPIContextProvider (props: PropsWithChildren<{}>): JSX.Elemen
...
+
+        const user = await getUser()
+        let urlParams
+        if ((user.mail.length > 0) && (user.mobileNumber.length > 0)) {
+          const params = { mail: user.mail, phone: user.mobileNumber }
+          urlParams = new URLSearchParams(params).toString()
+        }
+
+        const baseUrl = getEnvironment(Updates.releaseChannel).dfxPaymentUrl
+        const url = `${baseUrl}/kyc?code=${user.kycHash}${(urlParams != null) ? '&' + urlParams : ''}`
```
Comment (davidleomay) [inline shared/contexts/DFXAPIContextProvider.tsx:104]: WDYT about something like this?
```ts
const params = {
   code: user.kycHash,
};
if (user.mail?.length > 0) params.mail = user.mail;
if (user.mobileNumber?.length > 0) params.phone = user.mobileNumber;

const url = `${baseUrl}/kyc?{new URLSearchParams(params).toString()}`;
```

File: shared/api/dfx/ApiService.ts (hunk 1 above)
Comment (ThaBrad) [reply shared/api/dfx/ApiService.ts:90]: As I see no changes in Swagger to the code and non-code version, I kinda **have** to leave this in for now 🤷‍♂️ .

I can resort them, when they have been updated (as we discussed, we wanted to keep some legacy calls, like _post_).

**Review by davidleomay — APPROVED** [review]

File: shared/api/dfx/ApiService.ts (hunk 1 above)
Comment (davidleomay) [reply shared/api/dfx/ApiService.ts:90]: Don't really understand the argument, but I'm fine if we resort this later 😉

**Pull request merged**